src/data.rs

```diff
@@ -450,6 +453,99 @@ pub fn decode_block_cid_ask_message(msg: Vec<u8>) -> Option<(i128, Option<Cid>)>
 	}
 }
 
+pub async fn ipfs_priority_get_cells(
+	cells: &mut Vec<DataCell>,
+	ipfs: &Ipfs<DefaultParams>,
+	rpc_url: &String,
+	block_num: u64,
+) -> Result<()> {
+	println!("ipfs priority get started...");
+	// Try fetching the cells from IPFS first
+	for (_, req_cell_data) in cells.iter_mut().enumerate() {
+		let peers = ipfs.peers();
+		println!("number of peers: {}", peers.len());
+		if peers.len() == 0 {
+			break;
+		}
+		// println!("list of connections: {:?}", ipfs.connections());
+		// TODO: paralelize fetching
+		// Skip cells that return error when fetching from IPFS
+		println!(
+			"Requesting from IPFS: CID: {}. Cell reference: {}.",
+			req_cell_data.cid(),
+			req_cell_data.reference()
+		);
+		let mut block_cid: Cid = Cid::default();
+		if let Ok(record) = ipfs
+			.get_record(
+				Key::from(req_cell_data.reference().as_bytes().to_vec()),
+				ipfs_embed::Quorum::One,
+			)
+			.await
+		{
+			block_cid = Cid::try_from(record[0].record.value.to_vec()).unwrap();
+		}
```

Comment:
If we want to handle get or parse record errors instead to panic, we could have (CID, Option<Error>) instead of unwrapping:

```
		let key = Key::from(req_cell_data.reference().as_bytes().to_vec());

		let (block_cid, get_record_error): (Cid, _) = ipfs
			.get_record(key, ipfs_embed::Quorum::One)
			.await
			.map(|record| record[0].record.value.to_vec())
			.and_then(|cid_value| Cid::try_from(cid_value).context("Invalid CID value"))
			.map(|cid| (cid, None))
			.unwrap_or_else(|error| (Cid::default(), Some(error)));

		if let Some(error) = get_record_error {
			log::error!("Cannot get CID from record: {}", error);
		}
```

which would prevent ignoring errors.

Reply:
This is indeed a better approach, thanks.